kr8s/_api.py

```diff
@@ -520,6 +520,8 @@ async def async_watch(
         ) as (obj_cls, response):
             async for line in response.aiter_lines():
                 event = json.loads(line)
+                if event["object"].get("code") == 410:
```

Comment:
This seems like a fine place to catch this, but I think it would be a bit more readable if we check the type of the object too. It might just be a little clearer what we are doing.

```suggestion
                if event["object"]["kind"] == "Status" and event["object"].get("code") == 410:
```